File: examples/output_to_file_and_read_lines/output_to_file_and_read_lines.pl
use strict;
use FileHandle;
my $output_file = "/tmp/output_perl.txt";
if(my $fh = new FileHandle('> '.$output_file) ){
  print $fh "Hello World!\n\nAdditional line\n";
  $fh->close();
}
else{
  die("Failed to write to ".$output_file);
}

if(my $fh = new FileHandle($output_file) ){
  my $cnt = 0;
  while(<$fh>){
    print $_;
    $cnt++;
  }
  $fh->close();
  print $cnt."\n";
}

File: examples/output_to_file_and_read_lines/output_to_file_and_read_lines.js
const fs = require('fs');
const readline = require('readline');
let output_file = "/tmp/output_js.txt";
let line_cnt = 0;
fs.writeFile(output_file, "Hello World!\n\nAdditional line\n", (err, resp) => {
  if (err) throw err;
  const fileStream = fs.createReadStream(output_file);
  const rl = readline.createInterface({
    input: fileStream,
    crlfDelay: Infinity
  });
  rl.on('line', (line) => {
    console.log(`${line}`);
    line_cnt++;
  });
  rl.on('close', ()=>{console.log(line_cnt);});
});
